File: scripts/uninstall-skill.py
#!/usr/bin/env python3
"""Safely preview or uninstall one skill from an agent skills directory."""

from __future__ import annotations

import argparse
import sys

from _skill_tools import SkillToolError, resolve_target, uninstall_skill


def build_parser() -> argparse.ArgumentParser:
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("skill", help="lowercase hyphenated skill name")
    target = parser.add_mutually_exclusive_group()
    target.add_argument("--agent", choices=("codex", "pi"), help="installation profile (default: codex)")
    target.add_argument("--target-dir", help="explicit skills directory; overrides agent profiles")
    parser.add_argument(
        "--apply",
        action="store_true",
        help="perform the uninstall; without this flag the command is a dry run",
    )
    return parser


def main(argv: list[str] | None = None) -> int:
    args = build_parser().parse_args(argv)
    try:
        target_root = resolve_target(args.agent, args.target_dir)
        result = uninstall_skill(args.skill, target_root, apply=args.apply)
    except (OSError, SkillToolError) as exc:
        print(f"error: {exc}", file=sys.stderr)
        return 1

    print(f"{result.action}: {args.skill}")
    print(f"destination: {result.destination}")
    if result.action == "would-uninstall":
        print("rerun with --apply to remove this exact destination")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: scripts/uninstall-skill.ts
import { argumentsFor, fail, targetConflict } from "./arguments.ts";
import { resolveTarget, uninstallSkill } from "./skill-tools.ts";
try {
  const { values, positionals } = argumentsFor({
    allowPositionals: true,
    options: {
      agent: { type: "string" },
      "target-dir": { type: "string" },
      apply: { type: "boolean" },
      help: { type: "boolean", short: "h" },
    },
  });
  if (values.help)
    console.log(
      "Usage: node --experimental-strip-types scripts/uninstall-skill.ts SKILL [--agent codex|pi | --target-dir DIRECTORY] [--apply]\nWithout --apply, only preview removal of this exact destination.",
    );
  else {
    targetConflict(values);
    if (positionals.length !== 1)
      throw new TypeError("Exactly one skill name is required");
    const result = uninstallSkill(
      positionals[0],
      resolveTarget(values.agent, values["target-dir"]),
      values.apply,
    );
    console.log(
      `${result.action}: ${positionals[0]}\ndestination: ${result.destination}`,
    );
    if (result.action === "would-uninstall")
      console.log("rerun with --apply to remove this exact destination");
  }
} catch (error) {
  fail(error);
}
